### ISSUE TITLE
CMake build next branch

### ISSUE THREAD

**Issue by bendudson** [boutproject/BOUT-dev#2273]
I'm trying to use CMake to build BOUT++ `next` branch, but am running into somewhat puzzling errors. Compiling as in the manual (https://bout-dev.readthedocs.io/en/latest/user_docs/installing.html#cmake)
```
$ git clone --branch next git@github.com:boutproject/BOUT-dev.git BOUT-test
$ cd BOUT-test
$ cmake . -B build
$ cmake --build build

$ cd examples/blob2d
$ cmake . -B build -DCMAKE_PREFIX_PATH=$HOME/github/BOUT-test/build
```
Produces an error
```
CMake Error at /usr/share/cmake-3.19/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
  find_package called with invalid argument ".."
Call Stack (most recent call first):
  /home/bd512/github/BOUT-test/build/bout++Config.cmake:107 (find_dependency)
  CMakeLists.txt:6 (find_package)
```

In the file `/home/bd512/github/BOUT-test/build/bout++Config.cmake:107` 
```
find_dependency(netCDFCxx ..)
```
Changing this to 
```
find_dependency(netCDFCxx )
```
produces
```
-- Found netCDFCxx: /usr/lib/libnetcdf_c++4.so (found version "..") 
-- Configuring done
CMake Error at /home/bd512/github/BOUT-test/cmake/BOUT++functions.cmake:31 (add_executable):
  Target "blob2d" links to target "Libuuid::libuuid" but the target was not
  found.  Perhaps a find_package() call is missing for an IMPORTED target, or
  an ALIAS target is missing?
Call Stack (most recent call first):
  /home/bd512/github/BOUT-test/cmake/BOUT++functions.cmake:53 (bout_add_model)
  CMakeLists.txt:9 (bout_add_example)
```

Any suggestions welcome

**Comment by bendudson** [comment]
Somehow the `netCDFCxx_ROOT` variable is being set to `..` but I'm not sure where. It doesn't seem to be set in any of the .cmake or CMakeLists files

**Comment by bendudson** [comment]
Ah sorry, it's `netCDFCxx_VERSION` which gets set to `..` and then `find_package` complains that it's an invalid version.

**Comment by ZedThree** [comment]
Oof. Probably the result of a not very robust grep in
cmake/FindNetCDFCxx.cmake

I think it greps a header to work out the version? I suspect it's not
populating the version correctly, and .. should be major.minor.patch but
they're all empty

**Comment by bendudson** [comment]
Thanks @ZedThree I've got a couple of fixes about to open a PR

**Comment by bendudson** [comment]
For some reason the version string `_ncxx4_version` is coming back from `bout_inspect_netcdf_config(_ncxx4_version "${NCXX4_CONFIG}" "--version")`  empty, even though running `ncxx4-config --version` gives an output in the expected format "netCDF-cxx4 4.3.1"

**Comment by ZedThree** [comment]
You can check in `build/CMakeCache.txt` that `NCXX4_CONFIG` looks correct, maybe there's something strange happening there?

**Comment by bendudson** [comment]
In `build/CMakeCache.txt` there's an entry:
```
//Path to netCDF C++ config helper
NCXX4_CONFIG:FILEPATH=/usr/bin/ncxx4-config
```
which is the correct executable. The `NCXX_CONFIG` variable is used as
```
bout_inspect_netcdf_config(_ncxx4_version "${NCXX4_CONFIG}" "--version")
```
which calls
```
function(bout_inspect_netcdf_config VAR NX_CONFIG ARG)
  execute_process(
    COMMAND ${NX_CONFIG} ${ARG}
    OUTPUT_VARIABLE NX_CONFIG_OUTPUT
    OUTPUT_STRIP_TRAILING_WHITESPACE
  )
  if(EXISTS "${NX_CONFIG_OUTPUT}")
    set(${VAR} ${NX_CONFIG_OUTPUT} PARENT_SCOPE)
  endif()
endfunction()
```

**Comment by ZedThree** [comment]
I can reproduce this on my machine now so I'll investigate and get it sorted. Sorry, I must've broken this and not noticed!

**Comment by ZedThree** [comment]
Fixed in #2275
